service/controller/release/resource/apps/resource.go

```diff
@@ -122,6 +124,21 @@ func (r *Resource) ensureState(ctx context.Context) error {
 	return nil
 }
 
+func (r *Resource) excludeUnusedDeprecatedReleases(releases releasev1alpha1.ReleaseList) releasev1alpha1.ReleaseList {
```

Comment:
It would be cleaner to accept and return `[]releasev1alpha1.Release` here since we're just using the `Items` field in the `ReleaseList`.

Reply:
From the caller side I think it is nicer to pass the releases around the same way and operate on the whole API type:
```go
releases = excludeDeletedRelease(releases)
releases = r.excludeUnusedDeprecatedReleases(releases)
```
as opposed to
```go
releases.Items = excludeDeletedRelease(releases.Items)
releases.Items = r.excludeUnusedDeprecatedReleases(releases.Items)
```
But let me know if you feel strongly

Reply:
I was thinking of extracting `Items` from the `ReleaseList` earlier on, but this is fine too.